src/Hazelcast.Net/Clustering/ClusterMembers.cs

```diff
@@ -291,20 +292,33 @@ private void LogDiffs(MemberTable table, Dictionary<MemberInfo, int> diff)
             {
                 msg.Append("    ");
                 msg.Append(m.ToShortString(true));
-                var status = d switch
+                string status;
+                switch (d)
                 {
-                    1 => "Removing",
-                    2 => "Adding",
-                    3 => "Unchanged",
-                    _ => ""
-                };
+                    case 1:
+                        status = "Removing";
+                        break;
+                    case 2:
+                        status = "Adding";
+                        break;
+                    case 3:
+                        status = "Unchanged";
+                        countOfUnchanged++;
+                        break;
+                    default:
+                        status = "";
+                        break;
+                }
+
                 msg.Append(' ');
                 msg.Append(status);
                 msg.AppendLine();
             }
             msg.Append('}');
 
-            _logger.LogInformation(msg.ToString());
+            //Print only if there is a change
+            if (countOfUnchanged != table.Count)
```

Comment:
can this happen? or is it only for safety reasons? like, we get 3 unchanged members and that is all *yet* the table contains 4 members, so probably something is wrong, and it's better to log?

Reply:
I couldn't find a way that can occur. diff and members are already converged while filling the diff dictionary but checked the count against to diff to be more strictive. Accepting the diff as a source of truth, and if all entries are unchanged then we can skip the logging.

Also, I realized that I checked the method call as AtMost(1) which could be false positive. Imroved that one too on https://github.com/hazelcast/hazelcast-csharp-client/commit/e3ec104b568f9c6dc411c5dcc0b35f0331c6e51f